cypress/integration/channels.js

```diff
@@ -0,0 +1,152 @@
+      .get(ADD_CHANNEL_FORM_SELECTOS.backToChannelsList)
+      .click()
+      .get(CHANNELS_SELECTORS.channelsTable)
+      .contains(randomChannel)
+      .visit(URL_LIST.dashbord)
+      .get(HEADER_SELECTORS.channelSelect)
+      .click()
+      .get(HEADER_SELECTORS.channelSelectList)
+      .contains(randomChannel)
+      .click()
+      .visit(URL_LIST.products)
+      .addAliasToRequest("InitialProductFilterData")
+      .wait("@InitialProductFilterData");
+    cy.get(PRODUCTS_SELECTORS.productsList)
+      .first()
+      .click()
+      .get(PRODUCTS_SELECTORS.availableManageButton)
+      .click()
+      .get(PRODUCTS_SELECTORS.channelsAvailabilityForm)
+      .contains(randomChannel);
+  });
+
+  it("should validate slug name", () => {
+    const randomChannel = `${channelStartsWith} ${faker.random.number()}`;
+    channels.createChannel(false, randomChannel, randomChannel, currency);
+    cy.visit(URL_LIST.channels);
+    createChannel(randomChannel);
+    cy.get(ADD_CHANNEL_FORM_SELECTOS.slugValidationMessage).should(
+      "be.visible"
+    );
+  });
+
+  it("should validate currency", () => {
+    const randomChannel = `${channelStartsWith} ${faker.random.number()}`;
+    cy.visit(URL_LIST.channels);
+    createChannel(randomChannel, "notExistingCurrency");
+    cy.get(ADD_CHANNEL_FORM_SELECTOS.currencyValidationMassege).should(
+      "be.visible"
+    );
+  });
+
+  it("should delete channel", () => {
+    const randomChannelToDelete = `${channelStartsWith} ${faker.random.number()}`;
+    channels.createChannel(
+      false,
+      randomChannelToDelete,
+      randomChannelToDelete,
+      currency
+    );
+    cy.visit(URL_LIST.channels)
+      .get(CHANNELS_SELECTORS.channelName)
+      .contains(randomChannelToDelete)
+      .parentsUntil(CHANNELS_SELECTORS.channelsTable)
+      .find("button")
+      .click()
+      .get(BUTTON_SELECTORS.submit)
+      .click()
+      .addAliasToRequest("Channels")
+      .wait("@Channels");
+    cy.get(CHANNELS_SELECTORS.channelName)
+      .contains(randomChannelToDelete)
+      .should("not.exist");
+  });
+
+  it("should not be possible to add products to order with inactive channel", () => {
+    const randomChannel = `${channelStartsWith} ${faker.random.number()}`;
+    channels.createChannel(false, randomChannel, randomChannel, currency);
+    cy.visit(URL_LIST.orders)
+      .get(ORDERS_SELECTORS.createOrder)
+      .click()
+      .get(CHANNEL_FORM_SELECTORS.channelSelect)
+      .click()
+      .get(CHANNEL_FORM_SELECTORS.channelOption)
+      .contains(randomChannel)
+      .click()
+      .get(CHANNEL_FORM_SELECTORS.confirmButton)
+      .click();
+    cy.location()
+      .should(loc => {
+        const urlRegex = new RegExp(`${URL_LIST.orders}/.+`, "g");
+        expect(loc.pathname).to.match(urlRegex);
+      })
+      .get(DRAFT_ORDER_SELECTORS.addProducts)
+      .should("not.exist");
+  });
+
+  function createChannel(name, otherCurrency, slug = name) {
```

Comment:
Such helper functions should be in separate utils file. 
Performance: creating channel using GUI (cy.click etc) should be done once in test for channel creation view. If you want to create channel to perform other tests on it, its better to just run direct API query - it will be faster.

Reply:
This function is used in three tests, only in one of them channel will be created. Two other tests are using it to try to create channel and in result validation messages will show up, so I can't do it through api. This function will be used only in this file, so I didn't want to move it to separate file. Maybe I should rename function to tryToCreateChannel() ?

Reply:
I agree with @krzysztofwolski as to moving it outside of this file (utils ?) - it's not a problem now, but once we add more tests and more utils functions it'll become more and more difficult to navigate through those files, since it's not a place for utils function to be in a test file.